src/main/java/com/example/onlinebookshop/mapper/ShoppingCartMapper.java

```diff
@@ -0,0 +1,36 @@
+package com.example.onlinebookshop.mapper;
+
+import com.example.onlinebookshop.config.MapperConfig;
+import com.example.onlinebookshop.dto.cartitem.response.CartItemDto;
+import com.example.onlinebookshop.dto.shoppingcart.response.ShoppingCartDto;
+import com.example.onlinebookshop.entities.ShoppingCart;
+import org.mapstruct.AfterMapping;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingTarget;
+
+@Mapper(config = MapperConfig.class)
+public interface ShoppingCartMapper {
+    @Mapping(target = "userId", ignore = true)
+    @Mapping(target = "cartItems", ignore = true)
+    ShoppingCartDto toShoppingCartDto(ShoppingCart shoppingCart);
+
+    @AfterMapping
+    default void setUserId(@MappingTarget ShoppingCartDto shoppingCartDto,
+                           ShoppingCart shoppingCart) {
+        shoppingCartDto.setUserId(shoppingCart.getUser().getId());
+    }
+
+    @AfterMapping
+    default void setItems(@MappingTarget ShoppingCartDto shoppingCartDto,
+                          ShoppingCart shoppingCart) {
+        shoppingCart.getCartItems().forEach(cartItem -> {
```

Comment:
To avoid this, we can create CartItemMapper and add `@Mapper(config = MapperConfig.class, uses = CartItemMapper.class)` under ShoppingCartMapper.
In CartItemMapper implement such method

```
@Mapping(source = "book.id", target = "bookId")
@Mapping(source = "book.title", target = "bookTitle")
CartItemDto toDto(CartItem cartItem);
```

And remove `@Mapping(target = "cartItems", ignore = true)`